src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -584,6 +584,82 @@ zocl_xclbin_refcount(struct drm_zocl_dev *zdev)
 	return zdev->zdev_xclbin->zx_refcnt;
 }
 
+/*
+ * Initialize the memory structure in zocl driver based on the memory
+ * topology extracted from xclbin.
+ *
+ * Currently, we could have multiple memory sections but only two type
+ * of them could be marked as used. We identify the memory type by its
+ * tag. If the tag field contains "MIG", it is PL-DDR. Other tags
+ * e.g. "HP", "HPC", it is CMA memory.
+ *
+ * PL-DDR is managed by DRM MM Range Allocator;
+ * CMA is managed by DRM CMA Allocator.
+ */
+static void
+zocl_init_mem(struct drm_zocl_dev *zdev, struct mem_topology *mtopo)
+{
+	struct zocl_mem *memp;
+	int i;
+
+	zdev->num_mem = mtopo->m_count;
+	zdev->mem = vzalloc(zdev->num_mem * sizeof(struct zocl_mem));
+
+	for (i = 0; i < zdev->num_mem; i++) {
+		struct mem_data *md = &mtopo->m_mem_data[i];
+
+		if (!md->m_used)
+			continue;
+
+		memp = &zdev->mem[i];
+		if (md->m_type == MEM_STREAMING) {
+			memp->zm_type = ZOCL_MEM_TYPE_STREAMING;
+			continue;
+		}
+
+		memp->zm_base_addr = md->m_base_address;
+		/* In mem_topology, size is in KB */
+		memp->zm_size = md->m_size * 1024;
+		memp->zm_used = 1;
+
+		if (!strstr(md->m_tag, "MIG")) {
+			memp->zm_type = ZOCL_MEM_TYPE_CMA;
+			continue;
+		}
+
+		memp->zm_mm = vzalloc(sizeof(struct drm_mm));
+		memp->zm_type = ZOCL_MEM_TYPE_PLDDR;
+
+		drm_mm_init(memp->zm_mm, memp->zm_base_addr, memp->zm_size);
```

Comment:
The drm_mm_init() is DRM API. I don't think it is necessary to move init mem to xclbin.c

Reply:
Agree, let's discuss more about this, for now, I just revoke the changes in zocl_bo.c. The issue is that we still have #include xclbin.h elsewhere in zocl_bo.c and zocl_sysfs.c. We might have 2 choices: 1) include it once and everyone uses it; 2) using zocl_xclbin.h as the only interface to interact with xclbin internal data;